pkg/controller/gslb/dnsupdate.go

```diff
@@ -137,9 +137,15 @@ func (r *ReconcileGslb) gslbDNSEndpoint(gslb *k8gbv1beta1.Gslb) (*externaldns.DN
 		return nil, err
 	}
 
+	edgeDNSZone := os.Getenv("EDGE_DNS_ZONE")
+
 	for host, health := range serviceHealth {
 		var finalTargets []string
 
+		if !strings.Contains(host, edgeDNSZone) {
+			return nil, fmt.Errorf("ingress host %s is not matching delegated zone %s", host, edgeDNSZone)
```

Comment:
Grammar

```suggestion
			return nil, fmt.Errorf("ingress host %s does not match delegated zone %s", host, edgeDNSZone)
```

Reply:
@donovanmuller thanks a lot for fixing my Runglish, comrade ! :)